pkg/manager/runnable_group_test.go

```diff
@@ -224,3 +348,64 @@ var _ = Describe("runnableGroup", func() {
 		}
 	})
 })
+
+var _ LeaderElectionRunnable = &leaderElectionRunnableFunc{}
+
+// leaderElectionRunnableFunc is a helper struct that implements LeaderElectionRunnable
+// for testing purposes.
+type leaderElectionRunnableFunc struct {
+	RunFunc                func(context.Context) error
+	NeedLeaderElectionFunc func() bool
+}
+
+func (r leaderElectionRunnableFunc) Start(ctx context.Context) error {
+	return r.RunFunc(ctx)
+}
+
+func (r leaderElectionRunnableFunc) NeedLeaderElection() bool {
+	return r.NeedLeaderElectionFunc()
+}
+
+var _ WarmupRunnable = &warmupRunnableFunc{}
+
+// warmupRunnableFunc is a helper struct that implements WarmupRunnable
+// for testing purposes.
+type warmupRunnableFunc struct {
+	RunFunc    func(context.Context) error
```

Comment:
nit: Let's call this StartFunc, also in leaderElectionRunnableFunc & combinedRunnable